packages/vite/src/node/server/transformRequest.ts

```diff
@@ -228,42 +273,99 @@ async function loadAndTransform(
   const mod = await moduleGraph.ensureEntryFromUrl(url, ssr)
   ensureWatchedFile(watcher, mod.file, root)
 
-  // transform
-  const transformStart = isDebug ? performance.now() : 0
-  const transformResult = await pluginContainer.transform(code, id, {
-    inMap: map,
-    ssr
-  })
-  const originalCode = code
-  if (
-    transformResult == null ||
-    (isObject(transformResult) && transformResult.code == null)
-  ) {
-    // no transform applied, keep code as-is
-    isDebug &&
-      debugTransform(
-        timeFrom(transformStart) + colors.dim(` [skipped] ${prettyUrl}`)
-      )
-  } else {
-    isDebug && debugTransform(`${timeFrom(transformStart)} ${prettyUrl}`)
-    code = transformResult.code!
-    map = transformResult.map
-  }
+  let result: TransformResult | null = null
+
+  // persistent cache
 
-  if (map && mod.file) {
-    map = (typeof map === 'string' ? JSON.parse(map) : map) as SourceMap
-    if (map.mappings && !map.sourcesContent) {
-      await injectSourcesContent(map, mod.file, logger)
+  const persistentCacheKey = includedInPersistentCache
+    ? (_persistentCache?.getKey(id + code) ?? '') + (options.ssr ? '-ssr' : '')
+    : ''
+
+  if (includedInPersistentCache && !code.includes('import.meta.glob')) {
```

Comment:
This sync the check for reading and writing cache. If the version of TS is recent enough, `_persistentCache` should be non null given `includedInPersistentCache` is true.
```suggestion
  const persistentCacheKey = includedInPersistentCache &&  !code.includes('import.meta.glob')
    ? _persistentCache.getKey(id + code) + (options.ssr ? '-ssr' : '')
    : null

  if (persistentCacheKey) {
```

Reply:
Doesn't seem to be working :(

Reply:
I went with a different check